### ISSUE TITLE
关于在MSVD数据集上应用DSL的疑问

### ISSUE THREAD

**Issue by qjyyyy** [jpthu17/EMCL#3]
作者您好！最近看了您发表的论文，受益匪浅~作为同龄人真的很钦佩！

不知道您在EMCL中有没有试过在MSVD数据集上加DSL（inverted softmax），我尝试在这个数据集上加，但是发现DSL给的源代码是针对方阵的，也就是说文本和视频得是一对一的关系，而MSVD数据集在测试阶段是一个视频对应多个文本。不知道您有没有遇到过这个问题~~

**Comment by jpthu17** [comment]
抱歉这么晚才回复您。我认为在MSVD数据集上是可以加DSL的，但是我一直在忙其他的事情没时间去尝试。我觉得您可以试试在reshape sim matrix之前加DSL，这是t2v的处理代码，您可以试试:
```python
    t2v_sim_matrix = torch.from_numpy(sim_matrix).cuda()
    t2v_sim_matrix = t2v_sim_matrix * F.softmax(t2v_sim_matrix / args.t2v_temp, dim=0) * len(t2v_sim_matrix)
    sim_matrix = t2v_sim_matrix.cpu().numpy()
    if multi_sentence_:
        logger.info("before reshape, sim matrix size: {} x {}".format(sim_matrix.shape[0], sim_matrix.shape[1]))
        cut_off_points2len_ = [itm + 1 for itm in cut_off_points_]
        max_length = max([e_ - s_ for s_, e_ in zip([0] + cut_off_points2len_[:-1], cut_off_points2len_)])
        sim_matrix_new = []
        for s_, e_ in zip([0] + cut_off_points2len_[:-1], cut_off_points2len_):
            sim_matrix_new.append(np.concatenate((sim_matrix[s_:e_],
                                                  np.full((max_length - e_ + s_, sim_matrix.shape[1]), -np.inf)),
                                                 axis=0))
        sim_matrix_new = np.stack(tuple(sim_matrix_new), axis=0)
        logger.info("after reshape, sim matrix size: {} x {} x {}".
                    format(sim_matrix_new.shape[0], sim_matrix_new.shape[1], sim_matrix_new.shape[2]))

        tv_metrics = tensor_text_to_video_metrics(sim_matrix_new)
        vt_metrics = compute_metrics(tensor_video_to_text_sim(sim_matrix_new))
```

以上代码我并没有尝试跑过，所以可能没有效果......
